src/widget/form/groupchatform.cpp

```diff
@@ -234,6 +247,67 @@ void GroupChatForm::peerAudioPlaying(int peer)
     peerAudioTimers[peer]->start(500);
 }
 
+void GroupChatForm::resetLayout()
+{
+    if (!Settings::getInstance().getGroupPeerListSide())
+    {
+        if (mainSplitter)
+            layout()->replaceWidget(mainSplitter, chatWidget);
+
+        delete mainSplitter;
+        delete stringListModel;
+
+        namesListLayout = new FlowLayout(0,5,0);
+        headTextLayout->addLayout(namesListLayout);
+        onUserListChanged();
+    }
+    else
+    {
+        if (namesListLayout)
+        {
+            QLayoutItem *child;
+
+            while ((child = namesListLayout->takeAt(0)))
+            {
+                child->widget()->hide();
+                delete child->widget();
+                delete child;
+            }
+
+            peerLabels.clear();
+        }
+
+        delete namesListLayout;
+
+        QListView* listView = new QListView(this);
+        listView->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Expanding);
+        listView->setStyleSheet("QListView {background-color: white; border:none} QListView::item:selected {background-color:#6fc062;}");
+        listView->setEditTriggers(QAbstractItemView::NoEditTriggers);
+        listView->setSizeAdjustPolicy(QAbstractScrollArea::AdjustToContents);
+        stringListModel = new QStringListModel(group->getPeerList());
+        listView->setModel(stringListModel);
+
+        mainSplitter = new QSplitter(this);
+        mainSplitter->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+        layout()->replaceWidget(chatWidget, mainSplitter);
+        mainSplitter->setHandleWidth(6);
+        mainSplitter->addWidget(chatWidget);
+        mainSplitter->addWidget(listView);
+        mainSplitter->setStyleSheet("QSplitter::handle:horizontal {border-right: 1px solid #c1c1c1;}");
```

Comment:
I'll have to move that css to external file later, don't hardcode css in .cpp!

(it's a note to myself, not a comment)